include/libndt/libndt.hpp

```diff
@@ -1052,48 +937,6 @@ bool Client::run() noexcept {
       // subtests that we actually do not implement.
       return true;
     }
-    if (!connect()) {
-      LIBNDT_EMIT_WARNING("cannot connect to remote host; trying another one");
-      continue;
-    }
-    LIBNDT_EMIT_DEBUG("connected to remote host");
-    if (!send_login()) {
-      LIBNDT_EMIT_WARNING("cannot send login; trying another host");
-      continue;
-    }
-    LIBNDT_EMIT_DEBUG("sent login message");
-    if (!recv_kickoff()) {
-      LIBNDT_EMIT_WARNING("failed to receive kickoff; trying another host");
-      continue;
-    }
-    if (!wait_in_queue()) {
-      LIBNDT_EMIT_WARNING("failed to wait in queue; trying another host");
-      continue;
-    }
-    LIBNDT_EMIT_DEBUG("authorized to run test");
-    // From this point on we fail the test in case of error rather than
-    // trying with another host. The rationale of trying with another host
-    // above is that sometimes NDT servers are busy and we would like to
-    // use another one rather than creating queue at the busy one.
-    if (!recv_version()) {
-      return false;
-    }
-    LIBNDT_EMIT_DEBUG("received server version");
-    if (!recv_tests_ids()) {
-      return false;
-    }
-    LIBNDT_EMIT_DEBUG("received tests ids");
-    if (!run_tests()) {
-      return false;
-    }
-    LIBNDT_EMIT_DEBUG("finished running tests; now reading summary data:");
-    if (!recv_results_and_logout()) {
-      return false;
-    }
-    LIBNDT_EMIT_DEBUG("received logout message");
-    if (!wait_close()) {
-      return false;
-    }
     LIBNDT_EMIT_DEBUG("connection closed");
     return true;
```

Comment:
If you want to keep this code structure, maybe return `false` here and remove the debug message? If you are planning to refactoring this in a subsequent commit, then nevermind.

Reply:
Yes, the Locate v2 logic will replace all this. Your feedback is right. But, I'll defer changes to this replacement next.